### PR TITLE
Add an ``exclude`` option to exclude files based on ``glob`` patterns

### LINKED ISSUE #1: Add an ``exclude`` option
Should be able to exclude certain files or paths.

### PR DESCRIPTION
**Pull request by DanielNoord**
Closes #1 

`glob` is apparently pretty bad for ignoring stuff and is only used to capture things. (include vs. exclude).

However, for now I think this will be fine. I'm not sure about the tests and whether this catches everything, but at least we can get feedback if this works now!

### CODE REVIEW

File: tests/data/config/exclude_match/pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[tool.pydocstringformatter]
+exclude = "**/test_package/**"
```
Comment (Pierre-Sassoulas) [inline tests/data/config/exclude_match/pyproject.toml:2]: There's no newline at end of file there, maybe we could make the pre-commit exclude for newline and white space wider ?

**Review by Pierre-Sassoulas — APPROVED** [review]
Nice feature !

File: tests/data/config/exclude_match/pyproject.toml (hunk 1 above)
Comment (DanielNoord) [reply tests/data/config/exclude_match/pyproject.toml:2]: We could although I think stripping white spaces at the end of an docstring would also be a nice feature for us, so we'll need to then turn it off for the sub directory again?

Comment (Pierre-Sassoulas) [reply tests/data/config/exclude_match/pyproject.toml:2]: I think we should not step on the toes of pre-commit's ``trailing-whitespace`` and ``end-of-file-fixer``. But we can deactivate specifically test files for it like in pylint if we want to handle whitespaces. (https://github.com/PyCQA/pylint/blob/main/.pre-commit-config.yaml#L9)

Comment (DanielNoord) [reply tests/data/config/exclude_match/pyproject.toml:2]: Done with https://github.com/DanielNoord/pydocstringformatter/commit/71534deed2de0cd7254da5fc6699471a3a462d4b

**Pull request merged**